projects/pgai/tests/vectorizer/cli/test_vectorizer_document.py

```diff
@@ -0,0 +1,421 @@
+    vcr_: Any,
+):
+    """Test that a document is successfully embedded"""
+    connection = cli_db[1]
+    vectorizer_id = configure_document_vectorizer(cli_db[1], number_of_rows=len(docs))
+
+    # Huggingface.co hosts is being ignored globally.
+    # Docling will download its models if not found.
+    with vcr_.use_cassette("simple-docs.yaml"):
+        result = run_vectorizer_worker(cli_db_url, vectorizer_id)
+
+    if result.exception:
+        print(result.stdout)
+
+    with connection.cursor(row_factory=dict_row) as cur:
+        cur.execute("SELECT count(*) as errors FROM ai.vectorizer_errors;")
+        assert cur.fetchone()["errors"] == 0  # type: ignore
+
+        cur.execute("SELECT count(*) as count FROM documents_embedding_store;")
+        assert cur.fetchone()["count"] > len(docs)  # type: ignore
+
+        cur.execute("SELECT chunk FROM documents_embedding_store;")
+        chunks = cur.fetchall()
+        chunks_str = "\n".join([chunk["chunk"] for chunk in chunks])
+
+        # Sacred Texts of PostgreSQL.pdf
+        assert "And lo, there came forth PostgreSQL, blessed be its name" in chunks_str
+
+        # sample_pdf.pdf
+        assert "Maecenas mauris lectus" in chunks_str
+
+        # sample_with_table.pdf
+        assert "This is an example of a data table." in chunks_str
+
+        # sample_txt.txt
+        assert "Fromage frais cheese and biscuits danish fontina" in chunks_str
+
+        # basic-v3plus2.epub
+        # epub is not supported by docling but by pymupdf. auto parser handles this.
+        assert "Wants pawn term dare worsted ladle gull hoe lift" in chunks_str
+
+        # test.md
+        assert "Hello I am a test md document" in chunks_str
+
+        # lego_sets.pdf
+        assert "7190-1 Millennium Falcon" in chunks_str
+
+
+def test_simple_document_embedding_s3_no_credentials(
+    cli_db: tuple[TestDatabase, Connection],
+    cli_db_url: str,
+):
+    """Test that embedding fails when no credentials are available"""
+    vectorizer_id = configure_document_vectorizer(
+        cli_db[1], base_path="s3://adol-docs-test"
+    )
+
+    if "AWS_ACCESS_KEY_ID" in os.environ:
+        del os.environ["AWS_ACCESS_KEY_ID"]
+    if "AWS_SECRET_ACCESS_KEY" in os.environ:
+        del os.environ["AWS_SECRET_ACCESS_KEY"]
+
+    # This is necessary to prevent boto3 from using the default credentials file.
+    os.environ["AWS_SHARED_CREDENTIALS_FILE"] = "/dev/null"
+
+    # No cassette because it should never get to an API call.
+    result = run_vectorizer_worker(cli_db_url, vectorizer_id)
+
+    assert result.exit_code == 1
+    assert "Unable to locate credentials" in result.stdout
+
+
+def test_simple_document_embedding_s3(
+    cli_db: tuple[TestDatabase, Connection],
+    cli_db_url: str,
+    vcr_: Any,
+):
+    """Test that a document is successfully embedded"""
+    connection = cli_db[1]
+    vectorizer_id = configure_document_vectorizer(
+        cli_db[1], base_path="s3://adol-docs-test"
```

Comment:
Can we rename this bucket, so that it's more obvious that pgai's tests rely on this bucket existing? Or could we use something like minio instead, and not have a dependency on s3?

Reply:
Good point! I will use localstack as @MetalBlueberry suggested 💯

Reply:
We don't actually rely on this bucket for running the tests, since vcr has the request and response recorded. You can probably just rename it in the test and the recorded request yaml and it will still work.

Reply:
But you won't be able to recreate the VCR if we remove such a bucket. 
I ended up adding localstack via testcontainers as we did for other cases.